File: src/transformers/features/json/JsonIsParseTransformer.ts
import { JsonIsParseProgrammer } from "../../../programmers/json/JsonIsParseProgrammer";

import type { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace JsonIsParseTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "json.isParse",
      write: JsonIsParseProgrammer.write,
    });
}

File: packages/transform/src/features/json/JsonIsParseTransformer.go
package json

import (
	"github.com/samchon/typia/packages/core/src/programmers"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var JsonIsParseTransformer = jsonIsParseTransformerNamespace{}

type jsonIsParseTransformerNamespace struct{}

func (jsonIsParseTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return "(input => { try { const value = JSON.parse(input); return " + programmers.IsProgrammer.Write(call.TypeText) + "(value) ? value : null; } catch { return null; } })"
}
